test/test_tensordict.py

```diff
@@ -2935,6 +2935,52 @@ def test_unflatten_keys_collision(separator):
     assert assert_allclose_td(td.unflatten_keys(separator), ref)
 
 
+def test_split():
```

Comment:
Another version of this test would be to test with this setting:

Put the following tensordicts in a `pytest.mark.parametrize`:
```
    td = TensorDict({}, [10])
    td = TensorDict({"a": torch.zeros(10)}, [10])
    td = TensorDict({"a": torch.zeros(10, 5, 3), "b": torch.zeros(10, 5, 3, 4)}, [10, 5, 3])
```
and have a test where we do
```
for d in range(td.batch_dims):
    rep = td.shape[d] // 2
    split_list = [2] * rep + [td.shape[d] % 2]  % should return [2, 1] for size 3 and [2, 2, 1] for size 5
    for splits in (2, split_list):
        tds = td.split(splits, d)
```

etc

What do you think?

Reply:
Yeah I considered use `pytest.mark.parametrize` as well. 
I think pros are that it will make the test case reflect more of the options we provide in split_size (int or list(int)), but it would also make the test case less expressive and affect the readability, which is important in test files. What do you think?

Reply:
As sync-ed up offline, the test function will be moved to TestTensorDict class for wider coverage of test cases. Thanks for the feedback!